pyne/utils.py

```diff
@@ -216,31 +216,20 @@ def remove(path):
 
 def str_to_unicode(s):
     """
-    This function convert a str from binary or unicode to str (unicode).
-    If it is a list of string, convert every element of the list.
-
-    Parameters:
-    -----------
-    s : str or list of str
-
-    Returns:
-    --------
-    s : text str or list of unicode str
+    Convert str to unicode str.
     """
-    if isinstance(s, str) or isinstance(s, bytes):
-        # it is a str, convert to text str
-        try:
-            s = s.decode("utf-8")
-        except:
-            pass
+    if isinstance(s, bytes):
+        return s.decode()
+    elif isinstance(s, str):
         return s
+    elif isinstance(s, list):
+        return [str_to_unicode(x) for x in s]
+    elif isinstance(s, set):
+        return set(x if isinstance(x, str) else x.decode() for x in list(s))
```

Comment:
Why not invoke this function recursively as you've done in the other cases?

Reply:
Fixed in [6f281cd](https://github.com/pyne/pyne/pull/1472/commits/6f281cd77d546cb21dd31fdf34fe1cba39f7af58)
I have found that whenever I convert a list to a set, the values get rearranged, which can lead to incorrect results. However, I have found that using curly braces `{}` instead of `set()` fixes this issue.